otcextensions/osclient/dws/v1/cluster.py

```diff
@@ -366,7 +367,12 @@ def take_action(self, parsed_args):
         cluster = client.find_cluster(parsed_args.cluster)
         client.extend_cluster(cluster, parsed_args.add_nodes)
         if parsed_args.wait:
+            obj = client.get_cluster(cluster.id)
+            is_snapshotting = (obj.task_status == 'SNAPSHOTTING')
             client.wait_for_cluster(cluster.id, wait=parsed_args.timeout)
+            if is_snapshotting:
+                time.sleep(60)
+                client.wait_for_cluster(cluster.id, wait=parsed_args.timeout)
```

Comment:
i think you can add check on snapshooting into this func and then no need here additional checks

Reply:
i was also thinking about it, but i think It will be tricky to handle inside wait_For_cluster method